### PR TITLE
[WFCORE-2867] Credential store reload operation doesn't have effect on dependent resources

### PR DESCRIPTION
**Pull request by pskopek**
https://issues.jboss.org/browse/WFCORE-2867
https://issues.jboss.org/browse/JBEAP-10650

### CODE REVIEW

**Comment by pskopek** [comment]
retest this please

File: controller/src/main/java/org/jboss/as/controller/security/CredentialReference.java [hunk 1]
```diff
@@ -311,12 +312,16 @@ public static String credentialReferencePartAsStringIfDefined(ModelNode credenti
...
-        final InjectedValue<CredentialStore> credentialStoreInjectedValue = new InjectedValue<>();
+        final Service<CredentialStore> credentialStoreService;
         if (credentialAlias != null) {
             // use credential store service
             String credentialStoreCapabilityName = RuntimeCapability.buildDynamicCapabilityName(CREDENTIAL_STORE_CAPABILITY, credentialStoreName);
             ServiceName credentialStoreServiceName = context.getCapabilityServiceName(credentialStoreCapabilityName, CredentialStore.class);
-            serviceBuilder.addDependency(ServiceBuilder.DependencyType.REQUIRED, credentialStoreServiceName, CredentialStore.class, credentialStoreInjectedValue);
+            serviceBuilder.addDependency(ServiceBuilder.DependencyType.REQUIRED, credentialStoreServiceName);
+            ServiceController<?> controller = context.getServiceRegistry(false).getService(credentialStoreServiceName);
+            credentialStoreService = controller != null ? (Service<CredentialStore>)controller.getService() : null;
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/security/CredentialReference.java]: This is a race. There is no guarantee that when L321 executes the credential store service will be installed in MSC. Subsystems execute their operations in parallel during boot so subsystem A can in no way rely on subystem B having executed its ops first. And even without parallel boot there its totally valid for users to place the elytron subsystem last in their config.

File: elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreService.java [hunk 2]
```diff
@@ -112,8 +113,12 @@ public void start(StartContext startContext) throws StartException {
...
         try {
             credentialStoreAttributes.put(CS_LOCATION_ATTRIBUTE, loc.toAbsolutePath().toString());
-            credentialStore = getCredentialStoreInstance();
-            credentialStore.initialize(credentialStoreAttributes, resolveCredentialStoreProtectionParameter(), otherProviders.getOptionalValue());
+            CredentialStore cs = getCredentialStoreInstance();
+            cs.initialize(credentialStoreAttributes, resolveCredentialStoreProtectionParameter(), otherProviders.getOptionalValue());
+            if ( credentialStoreAttributes.get(ElytronDescriptionConstants.CREATE).equals("true") && !loc.toFile().exists() ){
+                cs.flush();
+            }
+            credentialStore.set(cs);
```
Comment (bstansberry) [inline elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreService.java]: In stop() should there be a credentialStore.set(null)?

Comment (pskopek) [inline elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreService.java]: Yes @bstansberry , you are right. Pushing update.

**Pull request merged**